plugins/image-prioritizer/helper.php

```diff
@@ -137,14 +142,170 @@ function image_prioritizer_add_element_item_schema_properties( array $additional
+				$response_code
+			)
+		);
+	}
+
+	// Validate that the Content-Type is an image.
+	$content_type = (array) wp_remote_retrieve_header( $r, 'content-type' );
+	if ( ! is_string( $content_type[0] ) || ! str_starts_with( $content_type[0], 'image/' ) ) {
+		return new WP_Error(
+			'background_image_response_not_image',
+			sprintf(
+				/* translators: %s is the content type of the response */
+				__( 'HEAD request for background image URL did not return an image Content-Type: %s.', 'image-prioritizer' ),
+				$content_type[0]
+			)
+		);
+	}
+
+	/*
+	 * Validate that the Content-Length is not too massive, as it would be better to err on the side of
+	 * not preloading something so weighty in case the image won't actually end up as LCP.
+	 * The value of 2MB is chosen because according to Web Almanac 2022, the largest image by byte size
+	 * on a page is 1MB at the 90th percentile: <https://almanac.httparchive.org/en/2022/media#fig-12>.
+	 * The 2MB value is double this 1MB size.
+	 */
+	$content_length = (array) wp_remote_retrieve_header( $r, 'content-length' );
+	if ( ! is_numeric( $content_length[0] ) ) {
+		return new WP_Error(
+			'background_image_content_length_unknown',
+			__( 'HEAD request for background image URL did not include a Content-Length response header.', 'image-prioritizer' )
+		);
+	} elseif ( (int) $content_length[0] > 2 * MB_IN_BYTES ) {
+		return new WP_Error(
+			'background_image_content_length_too_large',
+			sprintf(
+				/* translators: %s is the content length of the response  */
+				__( 'HEAD request for background image URL returned Content-Length greater than 2MB: %s.', 'image-prioritizer' ),
+				$content_length[0]
+			)
+		);
+	}
+
+	return true;
+}
+
+/**
+ * Filters the validity of a URL Metric with an LCP element background image.
+ *
+ * This removes the lcpElementExternalBackgroundImage from the URL Metric prior to it being stored if the background
+ * image URL is not valid. Removal of the property is preferable to
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param array<string, mixed>|mixed $data URL Metric data.
+ * @return array<string, mixed> Sanitized URL Metric data.
+ *
+ * @noinspection PhpDocMissingThrowsInspection
+ */
+function image_prioritizer_filter_store_url_metric_data( $data ): array {
+	if ( ! is_array( $data ) ) {
+		$data = array();
+	}
+
+	if ( isset( $data['lcpElementExternalBackgroundImage']['url'] ) && is_string( $data['lcpElementExternalBackgroundImage']['url'] ) ) {
+		$image_validity = image_prioritizer_validate_background_image_url( $data['lcpElementExternalBackgroundImage']['url'] );
+		if ( is_wp_error( $image_validity ) ) {
+			/**
+			 * No WP_Exception is thrown by wp_trigger_error() since E_USER_ERROR is not passed as the error level.
+			 *
+			 * @noinspection PhpUnhandledExceptionInspection
+			 */
+			wp_trigger_error( __FUNCTION__, $image_validity->get_error_message() . ' Background image URL: ' . $data['lcpElementExternalBackgroundImage']['url'] );
```

Comment:
Other error messages are translated, why not this one?

Reply:
I guess because I was being lazy.

Reply:
Ideal would be sprintf with a placeholder and a translator comment, no string concatenation.

So maybe something like

```php
sprintf(
	/* translators: 1: error message. 2: image url */
	__( 'Error: %1$s. Background image URL: %2$s., 'image-prioritizer' ),
	$image_validity->get_error_message(),
	$data['lcpElementExternalBackgroundImage']['url']
)
```

Reply:
See e1d0ac9dd935634b782d711c7e1ae85d296f44cf